packages/jest-runtime/src/index.ts

```diff
@@ -776,7 +778,46 @@ class Runtime {
       return this._environment.global.process;
     }
 
-    return require(moduleName);
+    const module = require(moduleName);
+
+    if (
+      !this._config.freezeCoreModules ||
+      this._config.freezeCoreModulesWhitelist.includes(moduleName)
+    ) {
+      return module;
+    }
+
+    if (this._coreModulesProxyCache[moduleName]) {
+      return this._coreModulesProxyCache[moduleName];
+    }
+
+    const set = (
+      target: object,
+      property: PropertyKey,
+      value: any,
+      receiver: any,
+    ) => {
+      if (typeof value !== 'function' || value._isMockFunction) {
```

Comment:
Add `target !== module` as additional case, such that the trap only triggers on the original module (see https://github.com/facebook/jest/pull/8331#issuecomment-578454230).